compiler/src/iree/compiler/DispatchCreation/HoistEncodingOps.cpp

```diff
@@ -88,26 +80,23 @@ bubbleUpSetEncodingThroughGenericOp(RewriterBase &rewriter,
   }
 
   RankedTensorType encodedType = encodingOp.getResultType();
-  AffineMap bcastMap = getBcastMapOrIdentity(rewriter, encodedType);
-  if (!bcastMap.isIdentity()) {
-    return rewriter.notifyMatchFailure(genericOp, "bcast_map map not identity");
-  }
-  if (outputMap.getNumDims() != bcastMap.getNumResults()) {
+  auto encoding = cast<IREE::Encoding::EncodingAttr>(encodedType.getEncoding());
+  AffineMap lastMap = encoding.getLastMapForOperandIndex();
```

Comment:
I'm refreshing my memory because I did not touch the code for a while. I was going to suggest using `getMapForOperandIndex` here. Because I think the idea of `user_indexing_map` is to track the mapping between whole iteration space and the iteration dimensions that the operand care. So I think we likely need `getMapForOperandIndex` for further propagation, and we can use it in the first place.

However, the current API only supports "appending indexing map", so it is not in the scope that we're working on. So it is fine to use `getLastMapForOperandIndex` here. We'll revisit this when we move the implementation to interfaces, so it is not a big deal to me. I care more about the encoding definition, atm.